ddtrace/tracer.py

```diff
@@ -884,11 +891,49 @@ def set_tags(self, tags):
         """
         self.tags.update(tags)
 
+    def _restore_from_shutdown(self):
+        with self._shutdown_lock:
+            if self.start_span is self._start_span:
+                # Already restored
+                return
+
+            atexit.register(self._atexit)
+
+            self.start_span = self._start_span
+            self.trace = self._trace
+
+            debtcollector.deprecate(
+                "Tracing with a tracer that has been shut down is being deprecated. "
+                "A new tracer should be created for generating new traces",
+                version="1.0.0",
+            )
```

Comment:
I would `log.warning` here ? since it is bad behavior/indicates a potential problem anyways?

not a strong opinion, this is actually totally fine too since it matches our upgrade checking/path for 1.0.0